python/cudaq/kernel/ast_bridge.py

```diff
@@ -2731,7 +2747,17 @@ def visit_Compare(self, node):
                 left = arith.FPToSIOp(comparator.type, left).result
             if IntegerType(left.type).width < IntegerType(
                     comparator.type).width:
-                left = arith.ExtSIOp(comparator.type, left).result
+                if IntegerType(left.type).width == 1:
```

Comment:
similar to above

Reply:
Applied the same change as recommended (5e1bc36df1f8a3000b32157fd723872f4fff05c2)